elements/lisk-p2p/src/validation.ts

```diff
@@ -159,13 +159,20 @@ export const checkProtocolVersionCompatibility = (
 	peerInfo: P2PDiscoveredPeerInfo,
 	nodeInfo: P2PNodeInfo,
 ): boolean => {
+	if (typeof nodeInfo.protocolVersion !== 'string') {
```

Comment:
in 1.4, protocolVersion will be `undefined`, so it will return here, but that should not happen

Reply:
Ah yep. I was meant to move it down but duplicated the whole if block instead :(

Reply:
This is checking `nodeInfo. protocolVersion` (our own node's protocolVersion). We only check `peerInfo.protocolVersion` below the other `if (!peerInfo.protocolVersion) {` block. So it should be OK.

Reply:
I will add a comment because it looks like accidentally duplicated code at a glance.